test/integration/targets/git/tasks/main.yml

```diff
@@ -291,6 +292,75 @@
     that:
       - 'git_result.stdout == "2cfde3668b8bb10fbe2b9d5cec486025ad8cc51b"'
 
+
+# Test that a forced shallow checkout referincing branch only always fetches latest head
+
+- name: clear checkout_dir
+  file: state=absent path={{ item }}
+  with_items:
+    - "{{ checkout_dir }}"
+    - "{{ checkout_dir }}.copy"
+
+- name: checkout example repo
+  git: repo={{ repo_format1 }} dest={{ checkout_dir }}
```

Comment:
Instead of using the external repo_format1 as base it would be better to setup a local repository. In the long term (nearly) all external dependencies should go away.

Just do `git init; echo; commit; ..`